build-tools/packages/build-cli/test/commands/typetests.test.ts

```diff
@@ -2,11 +2,162 @@
+				"broken-package": {
+					backCompat: false,
+					forwardCompat: false,
+				},
+			},
+			disabled: false,
+		},
+	};
+}
+
+/**
+ * A test package.json with a disabled typeValidation node.
+ */
+function packageWithTypeValidationDisabled(): PackageJson {
+	return {
+		...packageMinimal(),
+		devDependencies: {
+			"test-package-previous": "npm:test-package@2.0.0-internal.4.0.0",
+			"another-dependency": "^1.0.0",
+		},
+		typeValidation: {
+			broken: {
+				"broken-package": {
+					backCompat: false,
+					forwardCompat: false,
+				},
+			},
+			disabled: true,
+		},
+	};
+}
 
 describe("typetests tests", () => {
+	describe("applyTypeTestVersionOptions", () => {
+		describe("VersionOptions.Clear", () => {
+			it("removes previous test package dependency", () => {
+				const pkg = packageWithTypeValidation();
+				applyTypeTestVersionOptions(pkg, VersionOptions.Clear);
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+
+			it("removes previous test package dependency when type tests are disabled", () => {
+				const pkg = packageWithTypeValidationDisabled();
+				applyTypeTestVersionOptions(pkg, VersionOptions.Clear);
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+
+		describe("VersionOptions.ClearIfDisabled", () => {
+			it("leaves dependency when type tests are not disabled", () => {
+				const pkg = packageWithTypeValidation();
+				applyTypeTestVersionOptions(pkg, VersionOptions.ClearIfDisabled);
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.4.0.0",
+				);
+			});
+
+			it("removes previous test package dependency when type tests are disabled", () => {
+				const pkg = packageWithTypeValidationDisabled();
+				applyTypeTestVersionOptions(pkg, VersionOptions.ClearIfDisabled);
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+
+		describe("VersionOptions.Previous", () => {
+			it("sets previous version", () => {
+				const pkg = packageWithTypeValidation();
+				applyTypeTestVersionOptions(pkg, VersionOptions.Previous);
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.5.0.0",
+				);
+			});
+
+			it("sets previous version even without typeValidation node", () => {
+				const pkg = packageMinimal();
+				applyTypeTestVersionOptions(pkg, VersionOptions.Previous);
+				expect(pkg.devDependencies?.["test-package-previous"]).to.equal(
+					"npm:test-package@2.0.0-internal.5.0.0",
+				);
+			});
+
+			it("removes previous test package dependency when type tests are disabled", () => {
+				const pkg = packageWithTypeValidationDisabled();
+				applyTypeTestVersionOptions(pkg, VersionOptions.Previous);
+				expect(pkg.devDependencies?.["test-package-previous"]).not.to.exist;
+			});
+		});
+	});
+
+	describe("resetBrokenTests", () => {
+		it(`ignores packages with no typeValidation node`, () => {
```

Comment:
I'm surprised we don't have linting which prevents use of `` formatting when "" would work. Its a little odd to be inconsistent with this, but it doesn't really matter.

Reply:
This is a byproduct of refactoring. I'll clean it up.